src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateClone, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete stateClone[key];
+        }
+        break;
+
+      case 'clear':
+        for (const key of Object.keys(stateClone)) {
+          delete stateClone[key];
+        }
+        break;
+    }
+    result.push(Object.assign({}, stateClone));
```

Comment:
You can use spread syntax instead. But your variant is also correct. So, it's up to you 
```suggestion
    result.push({ ...stateClone });
```